src/databricks/labs/lsql/dashboards.py

```diff
@@ -183,19 +698,35 @@ def _format_query(query: str) -> str:
                     normalize=True,  # normalize identifiers to lowercase
                     pretty=True,  # format the produced SQL string
                     normalize_functions="upper",  # normalize function names to uppercase
-                    max_text_width=80,  # wrap text at 120 characters
+                    max_text_width=80,  # wrap text at 80 characters
                 )
             )
         formatted_query = ";\n".join(statements)
         return formatted_query
 
-    def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
-        """Create a dashboard from code, i.e. configuration and queries."""
-        dashboard_metadata = self._parse_dashboard_metadata(dashboard_folder)
-        widgets_metadata = self._get_widgets_metadata(dashboard_folder)
-        datasets = self._get_datasets(dashboard_folder)
-        widgets = self._get_widgets(widgets_metadata)
-        layouts = self._get_layouts(widgets, widgets_metadata)
+    def create_dashboard(
+        self,
+        dashboard_folder: Path,
+        *,
+        query_transformer: Callable[[sqlglot.Expression], sqlglot.Expression] | None = None,
+    ) -> Dashboard:
+        """Create a dashboard from code, i.e. configuration and queries.
+
+        Parameters :
+            dashboard_folder : Path
+                The path to the folder with dashboard files.
+            query_transformer : Callable[[sqlglot.Expression], sqlglot.Expression] | None (default | None)
+                A sqlglot transformer applied on the queries (SQL files) before creating the dashboard. If None, no
+                transformation will be applied.
+
+        Source :
+            https://sqlglot.com/sqlglot/transforms.html
+        """
+        dashboard_metadata = DashboardMetadata.from_path(dashboard_folder / "dashboard.yml")
+        tiles_metadata = self._parse_tiles_metadata(dashboard_folder, dashboard_metadata)
+        tiles = self._get_tiles(tiles_metadata, query_transformer=query_transformer)
+        datasets = self._get_datasets(tiles)
+        layouts = self._get_layouts(tiles)
```

Comment:
```suggestion
        layouts = list(self._get_layouts(tiles))
```
BUG: how did it serialise an iterator?..